### ISSUE TITLE
Static properties prevents tree shaking when declaration & declarator are split into 2 statements

### ISSUE THREAD

**Issue by Andarist** [rollup/rollup#2562]
- Rollup Version: 0.67
- Operating System (or Browser): irrelevant
- Node Version: irrelevant

### How Do We Reproduce?

Using those snippets on repl - https://github.com/Rich-Harris/agadoo/issues/4

### Expected Behavior

```js
var Foo = () => {}
Foo.defaultProps = {}
export { Foo }
```
and
```js
var Foo
Foo = () => {}
Foo.defaultProps = {}
export { Foo }
```
be tree-shaken in the same way.

### Actual Behavior

First gets tree shaken, latter does not.

**Comment by shellscape** [comment]
Hey folks. This is a saved-form message, but rest assured we mean every word. The Rollup team is attempting to clean up the Issues backlog in the hopes that the active and still-needed, still-relevant issues bubble up to the surface. With that, we're closing issues that have been open for an eon or two, and have gone stale like pirate hard-tack without activity.

We really appreciate the folks have taken the time to open and comment on this issue. Please don't confuse this closure with us not caring or dismissing your issue, feature request, discussion, or report. The issue will still be here, just in a closed state. If the issue pertains to a bug, please re-test for the bug on the latest version of Rollup and if present, please tag @shellscape and request a re-open, and we'll be happy to oblige.

---

Also of note, I ran those two snippets through the REPL with the latest version and the two results are functionally identical, but "worded" in different ways with a single line difference (not counting whitespace).

**Comment by Andarist** [comment]
@shellscape  I've retested this on REPL now and the results seems to be as reported in the past:
- [working](https://rollupjs.org/repl/?version=1.19.4&shareable=JTdCJTIybW9kdWxlcyUyMiUzQSU1QiU3QiUyMm5hbWUlMjIlM0ElMjJtYWluLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmV4cG9ydCUyMCU3QkJhciU3RCUyMGZyb20lMjAnLiUyRm1vZC5qcyclMjIlMkMlMjJpc0VudHJ5JTIyJTNBdHJ1ZSU3RCUyQyU3QiUyMm5hbWUlMjIlM0ElMjJtb2QuanMlMjIlMkMlMjJjb2RlJTIyJTNBJTIydmFyJTIwRm9vJTIwJTNEJTIwKCklMjAlM0QlM0UlMjAlN0IlN0QlNUNuRm9vLmRlZmF1bHRQcm9wcyUyMCUzRCUyMCU3QiU3RCU1Q252YXIlMjBCYXIlMjAlM0QlMjAoKSUyMCUzRCUzRSUyMCU3QiU3RCU1Q25leHBvcnQlMjAlN0IlMjBGb28lMkMlMjBCYXIlMjAlN0QlMjIlMkMlMjJpc0VudHJ5JTIyJTNBZmFsc2UlN0QlNUQlMkMlMjJvcHRpb25zJTIyJTNBJTdCJTIyZm9ybWF0JTIyJTNBJTIyY2pzJTIyJTJDJTIybmFtZSUyMiUzQSUyMm15QnVuZGxlJTIyJTJDJTIyYW1kJTIyJTNBJTdCJTIyaWQlMjIlM0ElMjIlMjIlN0QlMkMlMjJnbG9iYWxzJTIyJTNBJTdCJTdEJTdEJTJDJTIyZXhhbXBsZSUyMiUzQW51bGwlN0Q=)
- [not working](https://rollupjs.org/repl/?version=1.19.4&shareable=JTdCJTIybW9kdWxlcyUyMiUzQSU1QiU3QiUyMm5hbWUlMjIlM0ElMjJtYWluLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmV4cG9ydCUyMCU3QkJhciU3RCUyMGZyb20lMjAnLiUyRm1vZC5qcyclMjIlMkMlMjJpc0VudHJ5JTIyJTNBdHJ1ZSU3RCUyQyU3QiUyMm5hbWUlMjIlM0ElMjJtb2QuanMlMjIlMkMlMjJjb2RlJTIyJTNBJTIydmFyJTIwRm9vJTVDbkZvbyUyMCUzRCUyMCgpJTIwJTNEJTNFJTIwJTdCJTdEJTVDbkZvby5kZWZhdWx0UHJvcHMlMjAlM0QlMjAlN0IlN0QlNUNudmFyJTIwQmFyJTIwJTNEJTIwKCklMjAlM0QlM0UlMjAlN0IlN0QlNUNuZXhwb3J0JTIwJTdCJTIwRm9vJTJDJTIwQmFyJTIwJTdEJTIyJTJDJTIyaXNFbnRyeSUyMiUzQWZhbHNlJTdEJTVEJTJDJTIyb3B0aW9ucyUyMiUzQSU3QiUyMmZvcm1hdCUyMiUzQSUyMmNqcyUyMiUyQyUyMm5hbWUlMjIlM0ElMjJteUJ1bmRsZSUyMiUyQyUyMmFtZCUyMiUzQSU3QiUyMmlkJTIyJTNBJTIyJTIyJTdEJTJDJTIyZ2xvYmFscyUyMiUzQSU3QiU3RCU3RCUyQyUyMmV4YW1wbGUlMjIlM0FudWxsJTdE)

**Comment by shellscape** [comment]
Thanks for taking a second look. @lukastaegert I'm not sure it's the same issue as initially reported, but it sure doesn't look right. Can you have a quick look?

**Comment by Andarist** [comment]
Could you reopen the issue in such a case? 😉

**Comment by lukastaegert** [comment]
Yes it is. The reason is that anytime a variable is reassigned, e.g. because it is first declared without value and then later has a value assigned, it value is assumed to be unknown and adding properties to something unknown is a side-effect.

This might be fixed eventually but this will be a really long-term prospect.
